From 173b4246c6950973edffa0107c8b7b9f7dbdac39 Mon Sep 17 00:00:00 2001
From: Zahari Petkov <zahari@balena.io>
Date: Mon, 4 Feb 2019 21:33:53 +0100
Subject: [PATCH 1/2] Bring back ndo_do_ioctl fallback

---
 net/wireless/wext-core.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/net/wireless/wext-core.c b/net/wireless/wext-core.c
index 6cdb054484d6..28e5c4c24554 100644
--- a/net/wireless/wext-core.c
+++ b/net/wireless/wext-core.c
@@ -956,6 +956,11 @@ static int wireless_process_ioctl(struct net *net, struct iwreq *iwr,
 		else if (private)
 			return private(dev, iwr, cmd, info, handler);
 	}
+
+	if (dev->netdev_ops->ndo_do_ioctl) {
+		return dev->netdev_ops->ndo_do_ioctl(dev, (struct ifreq *) iwr, cmd);
+	}
+
 	return -EOPNOTSUPP;
 }
 
-- 
2.17.1

