From 0c7d639a5b133793c816a56fe93cc897f0ab8a31 Mon Sep 17 00:00:00 2001
From: Florin Sarbu <florin@balena.io>
Date: Tue, 22 Jul 2025 05:29:54 +0000
Subject: [PATCH] Use modules_install as wanted by yocto

Upstream-Status: Pending
Signed-off-by: Sebastian Panceac <sebastian@resin.io>
Signed-off-by: Florin Sarbu <florin@balena.io>
---
 Makefile | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Makefile b/Makefile
index 8d330f6..c348f33 100644
--- a/Makefile
+++ b/Makefile
@@ -40,3 +40,6 @@ all:
 clean:
 	$(MAKE) -C $(KDIR) M=$(PWD) clean
 	rm -f $(piControl-y)
+
+modules_install:
+	$(MAKE) ARCH=$(ARCH) CROSS_COMPILE="$(CROSS_COMPILE)" -C $(KDIR) M=$(PWD)  modules_install
-- 
2.34.1

